### ISSUE TITLE
We are supposed to have a better random chess rules.

### ISSUE THREAD

**Issue by Wingdinggaster656** [Monika-After-Story/MonikaModDev#7284]
See #6973.
I left a comment there.
Let me paste that comment here:

_I don't really think this is a bug, from a professional view._

_You're talking about losing instantly in random chess (the point is random), right?_

_In fact, this instantly lose is not a bug at all. We can imagine an extreme situation: all your pieces are randomly selected as pawns, while all the other pieces are randomly selected as queens, so the opponent immediately sends a queen to attack the pawns next to your king at the beginning of the game, ok, you lose. It's not a bug at all._

_While this is an extreme case, I can actually think of a large number of ways in which you could be checkmated instantly. For example, if your king is surrounded by knights and your opponent randomly gets two rooks in the same file as your king, you lose, too._

_If you want me to tell you what the problem is, the real problem is that random chess was never supposed to work this way. The orthodox random chess game is called <Chess 960>, and its rules don't simply random all the pieces. You can check out the rules on Google._

_The orthodox random chess is impossible to lose the first round immediately. What's more, winning or losing a game is not just a matter of luck in those rules(as opposed to chess today, where winning or losing a game depends almost entirely on the randomness of the opening game). After all, in openings, just a tiny advantage is enough to create an overwhelming attack._

_The random chess we have now is completely irrational. I have never seen any chess players play this mode with each other, and there are no websites that offer it._

**Comment by Wingdinggaster656** [comment]
About <Chess 960>, let me post some information here:

The idea for this variant came from former World Chess Champion Bobby Fischer, so it is also called the Fischer Chess or the Fischer Random Chess, but this Random change in the position of the initial pieces is determined by chance rather than by the game's necessity.
In Chess 960, at the beginning of each game, the initial positions of the pieces are randomly determined, namely, the positions of the king, queen, rook, horse, and elephant, which are not necessarily the same positions as in classic Chess. There are 960 different starting positions. If we introduce some rules in chess, such as the king must be placed in the middle of two rookies so that the king can change positions. Then we tried to record the opening of all the different initial permutations in 960. This is very important!
This is why a good memory of the opening order is not very important in chess. In other words, what really matters is the chess itself. But of course, the opening of a good player in classic chess also exists in Chess 960. That's because the same kind of analysis of the position of the pieces in chess, tactics and strategies, are just as valid here, just with the initial layout.
Fans of orthodox Chess will also like the Chess 960, because the 518th initial layout is the original layout of a classic Chess game.
Mathematically, orthodox Chess is just one part of Chess 960.

From this piece of information, we can see that the random game of Chess 960 practiced a very good mental pattern, and let people get rid of repeated openings (international masters estimate that they played 10,000 Sicilian defences, 7,000 Italian openings, those repetitive moves are really annoying), rather than the current situation, which is a game of luck.

**Comment by Wingdinggaster656** [comment]
The rules of chess960 is:
1.The black and white pieces are arranged symmetrically.
2.The two bishops cannot be placed in the same color square.
3.The two rooks must be on both sides of the king.
4.The pawns must be on the second and seventh horizontal lines(Which means, pawns are not randomized.)

**Comment by ThePotatoGuy** [comment]
we're looking to `Really Bad Chess` as the inspiration, not chess 960.

**Comment by Wingdinggaster656** [comment]
> we're looking to `Really Bad Chess` as the inspiration, not chess 960.

You mean the new name of the current Random Chess?
Yes, I think that's cool. This current version is replying on luck, but it's fun sometimes, this is true.
But do you guys mind if I made a chess 960 and start a pull request for that? If you guys are okay with that, I am going to make chess 960 in MAS some day later.

**Comment by ThePotatoGuy** [comment]
[Really Bad Chess is a real game](https://en.wikipedia.org/wiki/Really_Bad_Chess).

Yes, you are free to make a PR with a chess 960 implementation.

**Comment by Ant9cp** [comment]
I have a question and answer honestly 

I'm sooo bad at chess and i never ever won a game of chess before and i don't even know exactly how chess works

i've played chess against monika for like few times a month i think and i just don't understand what to do

Should i stop playing chess? because i just can't win a chess game 

i personally don't like keep playing a game where i literally keep losing and never win one's

**Comment by kkrosie123** [comment]
The thing about playing chess with Monika is that she is an AI. AIs will always make the *perfect* move, and so they are generally harder to play against than real people.

**Comment by Retrolovania** [comment]
Chess is a game that involves strategy. It can be quite difficult at times.
Heck, I haven't won any games against Monika, either. 
Like all things, Chess is something that requires practice to get better at it.

**Comment by Ant9cp** [comment]
Trust me if i play chess against someone in person for real the same would happen

**Comment by Retrolovania** [comment]
Well, that's how Chess is; hard, frustrating even. You need to keep playing it and improve your skills.

**Comment by Ant9cp** [comment]
Well i doubt that i will improve

**Comment by multimokia** [comment]
Honestly to be blunt, if your attitude is that you'll never improve and you'll always lose, don't play. You've already convinced yourself you won't win and as such won't put in the effort to do so.

Now if you're willing to spend the time to practice, review your games with an analyzer, and understand where you could have done something different, then by all means, play and keep learning from yourself.

But it takes that self initiative and confidence.

**Comment by Wingdinggaster656** [comment]
From the perspective of a professional player who has been playing chess for many, many years, I will tell you that chess is definitely not difficult now.

The average level-rating for the current chess is around 1,400 (assuming you lose as much as you win) and goes as low as about 1,000.
Let me tell you what 1400 is.

1400 means that even if you are a VERY untalented person, you can study for two or three months to reach that number. A normal person can even do it in less than a month. Someone with talent? It could be a week.

And then I'll tell you what 1000 is.
1000 is basically a beginner's level. Monika at this level will frequently make inaccurate moves and make various wrong moves.
If you really set your mind to study, you can achieve 1000 points in less than a month. That's the assumption that you don't have talent. The average is one week.

So for those of you who have never won, even in the current version, ask yourself, have you ever had a serious chess lesson? Have you ever been to a site like chess.com to get tutorials? Or at least ask your chess-playing friend?

Is it because you never study or because Monika is too good?

If you really don't know where to find a tutorial, great, that's fine, I'm making a series of chess tutorials for MAS. It's expected to be finished next month. You can come to this PR ahead of time to pick up some classes.

**Comment by Ant9cp** [comment]
So multimokia so what are you saying exactly sounds like you got a attitude

**Comment by Retrolovania** [comment]
...okay, that's just unnecessary.

**Comment by multimokia** [comment]
> I have a question and answer honestly

Hey man, don't ask for an honest response if you don't want the truth. I'm just saying, you have to dedicate time and work towards improving. If you're not willing to put in the effort, then there's just no point.

**Comment by Ant9cp** [comment]
I just want a yes or no to my question i just forgot to say this

**Comment by Wingdinggaster656** [comment]
> I just want a yes or no to my question i just forgot to say this
> No need to say what you just told me

My personal advice is, don't play for a while. 
You can learn those lessons after I finish them.
By the way, I've actually finished part of those lessons, why don't you go download the files and have a look?

**Comment by Olliesama** [comment]
https://i.kym-cdn.com/photos/images/newsfeed/000/771/166/3b4.png

**Comment by multimokia** [comment]
> By the way, I've actually finished part of those lessons, why don't you go download the files and have a look?

Please don't advise this. If changes are made to the event(s) themselves, they will not be changed for their install, meaning they can be very broken and potentially cause crashes/etc. when it's fully released.

**Comment by Ant9cp** [comment]
That image supposed to me?

**Comment by Wingdinggaster656** [comment]
> > By the way, I've actually finished part of those lessons, why don't you go download the files and have a look?
> 
> Please don't advise this. If changes are made to the event(s) themselves, they will not be changed for their install, meaning they can be very broken and potentially cause crashes/etc. when it's fully released.

Understood, will edit my comment. Sorry for the trouble I made.

**Comment by lunulae** [comment]
> Yes to be honest i should try harder but chess cannot be for everyone

Chess isn't for everyone, this is true. 

I would bet that a good number of MAS players have not gotten a win yet, and that's okay. If you don't enjoy learning chess, you don't have to play chess. There's nothing hidden behind a win, no achievement to unlock.

I have noticed, though, that you often come into the github to talk about your frustration with your losses, and some of these times you have taken out your frustration on devs, saying things like you don't believe they made any changes. This time, you brought it into a discussion about someone else's dev project.

I don't fault you for venting, but I'm not sure what response we're supposed to give you at this point. If you are frustrated and haven't got an interest in improving, it's okay to take a break or stop. It's not an interest everyone has to have, it's just one minigame in this mod.

TL;DR you're going to be okay if this isn't your interest. There will be other games, no one is going to be upset at you if there's one you decide not to play. It's your choice. But please don't take it out on devs, they're doing their best.

**Comment by Ant9cp** [comment]
Okay i have overreacted and i apologize for that, but i'm really upset about that image if that is a joke or something then i don't like that kind of humor i take things like that very seriously i have ptsd

**Comment by Wingdinggaster656** [comment]
> Okay i have overreacted and i apologize for that, but is that image supposed to be me dead or something? because i'm really upset about that image if that is a joke or something then i really don't like that kind of humor i take things like that very seriously i have ptsd and i can get a little sensitive
> 
> and if that image isn't a joke well then....

I think you are really sensitive this time. That image is obviously just a meme. It's not pointing at anyone. Don't worry.

**Comment by Wingdinggaster656** [comment]
Linked pull request of this issue:#7288
I've finished the Chess960 System there. I guess we have a better random chess rule now. And, devs are right, the old random chess rule is at least some of fun, so I don't think we should delete the old one too.
But on the other hand, let us just add the 960 chess, because this is the "universal" rule of random chess.
This issue will be closed by me when that pull request is merged or rejected.

**Comment by ThePotatoGuy** [comment]
[closing as implemented]
